maintenance/log/sink.go

```diff
@@ -0,0 +1,98 @@
+package log
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/hlog"
+	"github.com/rs/zerolog/log"
+)
+
+type sinkKey struct{}
+
+// WithSink wraps the given context in a new context with
+// the given Sink stored as value.
+func WithSink(ctx context.Context, sink *Sink) context.Context {
```

Comment:
I think a better naming would be `ContextWithSink`. Because `log.WithSink` sounds more like you are modifying a logger.